File: Medium/Python/ArrayAndStrings/LongestPalindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
            :type s: str
            :rtype: str
            """
        helper = ''
        result = ''
        for i in range(len(s)):
            # Find the palindromic substring like 'bab'
            helper = self.finder(s, i, i)
            if len(helper) > len(result):
                result = helper
            
            # Find the palindromic substring like 'abba'
            if i < len(s) - 1:
                helper = self.finder(s, i, i + 1)
                if len(helper) > len(result):
                    result = helper
        
    return result

    def finder(self, s, l, r):
        while l >= 0 and r < len(s) and s[l] == s[r]:
            l -= 1
            r += 1
        return s[l + 1: r]

File: Medium/Java/ArrayAndStrings/LongestPalindrome.java
class Solution {
    public String longestPalindrome(String s) {
        String res = "";
        for (int i = 0; i < s.length(); i ++) {
            //             find palindrome like 'aba'
            int[] temp = findPalindrome(s, i, i);
            if (temp[1] - temp[0] > res.length())
                res = s.substring(temp[0], temp[1]);
            
            //             find palindrome like 'abba'
            temp = findPalindrome(s, i, i + 1);
            if (temp[1] - temp[0] > res.length())
                res = s.substring(temp[0], temp[1]);
        }
        return res;
    }
    
    public int[] findPalindrome(String s, int l, int r) {
        while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            l --; r ++;
        }
        int[] res = {l + 1, r};
        return res;
    }
}
